src/formatters/deleteResultFormatter.ts

```diff
@@ -82,6 +106,85 @@ export class DeleteResultFormatter implements Formatter<DeleteSourceJson> {
+    if (this.testLevel === TestLevel.NoTestRun || !this.result.response.runTestsEnabled) {
+      ux.log();
+      return;
+    }
+
+    this.displayVerboseTestFailures();
+
+    if (this.verbosity === 'verbose') {
+      this.displayVerboseTestSuccesses();
+      this.displayVerboseTestCoverage();
+    }
+
+    ux.log();
+    ux.log(tableHeader('Test Results Summary'));
+    ux.log(`Passing: ${this.result.response.numberTestsCompleted ?? 0}`);
+    ux.log(`Failing: ${this.result.response.numberTestErrors ?? 0}`);
+    ux.log(`Total: ${this.result.response.numberTestsTotal ?? 0}`);
+    const time = this.result.response.details.runTestResult?.totalTime ?? 0;
+    if (time) ux.log(`Time: ${time}`);
+    // I think the type might be wrong in SDR
+    ensureArray(this.result.response.details.runTestResult?.codeCoverageWarnings).map(
+      (warning: CodeCoverageWarnings & { name?: string }) =>
+        ux.warn(`${warning.name ? `${warning.name} - ` : ''}${warning.message}`)
+    );
+  }
+
+  private displayVerboseTestCoverage(): void {
+    const codeCoverage = ensureArray(this.result.response.details.runTestResult?.codeCoverage);
+    if (codeCoverage.length) {
+      const coverage = codeCoverage.sort((a, b) => (a.name.toUpperCase() > b.name.toUpperCase() ? 1 : -1));
+      ux.log();
+      ux.log(tableHeader('Apex Code Coverage'));
+
+      ux.table(coverage.map(coverageOutput), {
+        name: { header: 'Name' },
+        numLocations: { header: '% Covered' },
+        lineNotCovered: { header: 'Uncovered Lines' },
+      });
+    }
+  }
+
+  private displayVerboseTestSuccesses(): void {
+    const successes = ensureArray(this.result.response.details.runTestResult?.successes);
+    if (successes.length > 0) {
+      const testSuccesses = sortTestResults(successes);
+      ux.log();
+      ux.log(success(`Test Success [${successes.length}]`));
+      for (const test of testSuccesses) {
+        const testName = underline(`${test.name}.${test.methodName}`);
+        ux.log(`${check} ${testName}`);
+      }
+    }
+  }
+
+  private displayVerboseTestFailures(): void {
+    if (!this.result.response.numberTestErrors) return;
+    const failures = ensureArray(this.result.response.details.runTestResult?.failures);
+    const failureCount = this.result.response.details.runTestResult?.numFailures;
+    const testFailures = sortTestResults(failures);
+    ux.log();
+    ux.log(error(`Test Failures [${failureCount}]`));
+    for (const test of testFailures) {
+      const testName = underline(`${test.name}.${test.methodName}`);
+      ux.log(`• ${testName}`);
+      ux.log(`  ${dim('message')}: ${test.message}`);
+      if (test.stackTrace) {
+        const stackTrace = test.stackTrace.replace(/\n/g, `${os.EOL}    `);
+        ux.log(`  ${dim('stacktrace')}: ${os.EOL}    ${stackTrace}`);
+      }
+      ux.log();
+    }
+  }
+
+  private determineVerbosity(): Verbosity {
+    if (this.flags.verbose) return 'verbose';
+    return 'normal';
+  }
+
```

Comment:
This is 99% copy/paste from https://github.com/salesforcecli/plugin-deploy-retrieve/blob/main/src/formatters/deployResultFormatter.ts#L299-L382

There's now 3 commands that report test failures:
```
project deploy start
project deploy validate
project delete source
```
`deploy validate` uses the `DeployResultFormatter` so it shares code with `deploy start` but `delete source` has a diff formatter.
One option to avoid duplicate code is to create a TestResultFormatter class and then `DeployResultFormatter` and `DeleteResultFormatter` extend it, what do you think @WillieRuemmele ? response from server seems to be identical but maybe I'm missing some scenario.

Reply:
I support that idea, it'll probably be similar to what plugin-source does/has

I believe the server response is the same, they're all "deploys" some are just only destructive changes, so I think the responses will be consistent between the two

Reply:
Done.